Task 1/Assets/Scripts/Generators/CuttingCalculator.cs

```diff
@@ -0,0 +1,35 @@
+using UnityEngine;
+
+namespace StackApp.Generators
+{
+    public class CuttingCalculator
+    {
+        public struct MeshesData
+        {
+            public Vector3 PlayablePosition;
+            public Vector3 FallingPosition;
+            public Vector3 PlayableSize;
+            public Vector3 FallingSize;
+        }
+
+        public MeshesData Calculate(Transform playableBlock, Transform movingBlock, Vector3 currentBlockSize)
```

Comment:
this method (class itself either) can be marked as static - as it doesn't store any state - and is just calculator